awx/main/fields.py

```diff
@@ -415,6 +415,13 @@ def from_db_value(self, value, expression, connection, context):
         return value
 
 
+@JSONSchemaField.format_checker.checks('vault_id')
+def format_vault_id(value):
+    if '@' in value:
+        raise jsonschema.exceptions.FormatError('@ is not an allowed character')
```

Comment:
We're building the `--vault-id <id>@prompt` for them; we only want them to type in the `<id>` part.